### PR TITLE
Add Chap 4, MAM

### PR DESCRIPTION
**Pull request by jkrvivian**

### CODE REVIEW

**Comment by jserv** [comment]
Did you rebase?

File: paper.tex [hunk 1]
```diff
@@ -287,17 +295,66 @@ \section{Masked Authenticate Messaging}
...
+\begin{figure}[!t]
+    \centering
+    \includegraphics[width=\linewidth]{delegation}
+    \caption{Data providers can use MAM in 2 ways: the provider performs MAM operations locally, and the provider delegate MAM operations to Tangle-accelerator.}
+    \label{fig:delegation}
+\end{figure}
+
+
+\subsubsection{Infrastructure}
+To build the system, MQTT is chosen as the communication protocol. MQTT protocol utilizes much more lightweight packet than HTTP protocol has. The topic-based publish-subscribe pattern and lightweight packet that MQTT adopts would undoubtedly play an important role on low-level edge devices with limited bandwidth. The publish-subscribe model decouples publishers and subscribers in many ways:
```
Comment (jserv) [inline paper.tex]: Cite MQTT.

File: paper.tex [hunk 2]
```diff
@@ -287,17 +295,66 @@ \section{Masked Authenticate Messaging}
...
+    \centering
+    \includegraphics[width=\linewidth]{MAM_E2EE}
+    \caption{The process of E2EE.}
+    \label{fig:MAM_E2EE}
+\end{figure*}
+
+In the Fig.~\ref{fig:MAM_E2EE}, we illustrate each step in the whole E2EE process we present. The following section addresses the process.
+
+\begin{enumerate}
+    \item Generate symmetric key with unique device identifier and Message Authentication Code (MAC) of the previous message. If the message is the genesis message, then a sharing secret initialization information would serve as the MAC of previous message. Key Derivation Function (KDF) is used to generate the symmetric keys.
```
Comment (jserv) [inline paper.tex]: s/key/keys/

**Comment by jkrvivian** [comment]
> Did you rebase?

yes!

File: paper.tex [hunk 3]
```diff
@@ -287,17 +295,66 @@ \section{Masked Authenticate Messaging}
...
 
-\subsection{TA - endpoint}
+\subsection{Delegate MAM operations to Tangle-accelerator}
 \label{section:ta_endpoint}
+MAM builds a secure and authenticated communication protocol on top of multiple cryptosystems, which low-level devices need to spend a lot of resources to apply it, and they may not even support built-in libraries in regular operating system. To overcome these concerns, we propose another approach that allows data providers to delegate the MAM operations to Tangle-accelerators, proxy servers with high computation power, which can accelerates the interactions with Tangle while ensuring the privacy of data providers. Fig.~\ref{fig:delegation} shows how data providers can adopt MAM.
+
+\begin{figure}[!t]
+    \centering
+    \includegraphics[width=\linewidth]{delegation}
+    \caption{Data providers can use MAM in 2 ways: the provider performs MAM operations locally, and the provider delegate MAM operations to Tangle-accelerator.}
```
Comment (jserv) [inline paper.tex]: s/delegate/delegates/

Comment (jserv) [reply paper.tex]: Did you have the diagram which illustrates the delegation between endpoint and TA?

File: paper.tex [hunk 4]
```diff
@@ -168,7 +169,11 @@ \subsection{Choice of Data Storage}
...
+MAM publishes authenticated streaming data to channel and endpoint as zero-value transactions to the Tangle and provides the ability to publish and fetch encrypted messages over the network along with data integrity and access control. The payload of a MAM message can be encrypted with an \textbf{encryption key} that restricts only authorized players can access contents, the ciphertext is then signed with \textbf{signature key} generated with MSS and attached to the Tangle. This approach allows users to validate the signatures without knowing the actual contents but ensuring the messages do come from the exact source. See Fig.~\ref{fig:channel_and_key} for illustration. 
+
+Using MAM as a data storage benefits from the scalability of the underlying IOTA network as well as the decentralized and fault-tolerant characteristic of distributed ledgers, which reduce the risks of centralized storage services. Furthermore, the rights of data access are traded instead of a copy of data in the trading platform, which eliminates the need for data consumers to have additional storage.
```
Comment (jserv) [inline paper.tex]: s/characteristic/characteristics/

File: paper.tex (hunk 3 above)
Comment (jkrvivian) [reply paper.tex]: yes, Fig.8, delegation.png

Comment (jkrvivian) [reply paper.tex]: Fig.9 has the workflow of E2EE.

File: paper.tex [hunk 5]
```diff
@@ -287,17 +295,66 @@ \section{Masked Authenticate Messaging}
...
+\begin{figure}[!t]
+    \centering
+    \includegraphics[width=\linewidth]{delegation}
+    \caption{Data providers can use MAM in 2 ways: the provider performs MAM operations locally, and the provider delegates MAM operations to Tangle-accelerator.}
+    \label{fig:delegation}
+\end{figure}
+
+
+\subsubsection{Infrastructure}
+To build the system, MQTT\cite{MQTT} is chosen as the communication protocol. MQTT protocol utilizes much more lightweight packet than HTTP protocol has. The topic-based publish-subscribe pattern and lightweight packet that MQTT adopts would undoubtedly play an important role on low-level edge devices with limited bandwidth. The publish-subscribe model decouples publishers and subscribers in many ways:
```
Comment (jserv) [inline paper.tex]: Replace the statement which mentioned "much more lightweight package" with "MQTT requires fewer TCP packets and less traffic."

**Review by jserv — CHANGES_REQUESTED** [review]
Fix conflicts

**Pull request merged**